sql/src/test/java/io/crate/execution/engine/collect/collectors/OrderedLuceneBatchIteratorFactoryTest.java

```diff
@@ -109,15 +128,44 @@ public void testOrderedLuceneBatchIterator() throws Exception {
                 LuceneOrderedDocCollector collector2 = createOrderedCollector(searcher2, 2);
                 return OrderedLuceneBatchIteratorFactory.newInstance(
                     Arrays.asList(collector1, collector2),
-                    OrderingByPosition.rowOrdering(new int[] { 0 }, reverseFlags, nullsFirst),
-                    MoreExecutors.directExecutor(),
+                    OrderingByPosition.rowOrdering(new int[]{0}, reverseFlags, nullsFirst),
+                    ROW_ACCOUNTING_SUPPLIER, MoreExecutors.directExecutor(),
                     true
                 );
             }
         );
         tester.verifyResultAndEdgeCaseBehaviour(expectedResult);
     }
 
+    @Test
+    public void testOrderedLuceneBatchIteratorTripsCircuitBreaker() {
+        RowAccounting rowAccounting = mock(RowAccounting.class);
+        Supplier<RowAccounting> rowAccountingSupplier = () -> rowAccounting;
+
+        doThrow(new CircuitBreakingException("tripped")).when(rowAccounting).accountForAndMaybeBreak(any(Row.class));
+
+        LuceneOrderedDocCollector collector1 = createOrderedCollector(searcher1, 1);
+        LuceneOrderedDocCollector collector2 = createOrderedCollector(searcher2, 2);
```

Comment:
are 2 collectors necessary for this test?

Reply:
ah, was focused on this use case that I miss the one where it's one _repeatable_ collector we should also wrap the iterator with the RamAccountingPagingIterator. Good catch! Thanks